xamarin/Xamarin.Forms
# Ios 2017 compatible Xamarin.Forms.Platform.iOS

### Description of Change ###
Build different ios platform dlls for xcode10 and xcode11 compatibility

The problem with our previous approach for Xcode10 compatibility is that at runtime or link time VS 2017 will encounter types that it doesn't understand. 

So given the scenario of a Xamarin.Forms.iOS application built on VS2017 running on iOS13 you will hit exceptions if you encounter types that VS2017 doesn't understand

This PR doesn't build our DLL on VS 2017 it only if defs out any apis that are iOS13 specific which should produce a dll that will compile on VS2017 and then run/link successfully

I've also included an extra iOS build lane that installs all the VS 2017 sdks and compiles our control gallery to ensure we aren't exposing any APIS that VS 2017 doesn't understand

### Platforms Affected ### 
- iOS

### Testing Procedure ###
- use this nuget with full linking on vs2017 and vs2019. Make sure everything runs

### PR Checklist ###
